src/Domain.LinnApps/Reports/SpendsReportService.cs

```diff
@@ -113,6 +118,65 @@ public ResultsModel GetSpendBySupplierReport(string vendorManagerId)
             return model;
         }
 
+        public ResultsModel GetSpendBySupplierByDateRangeReport(string fromDate, string toDate, string vendorManagerId)
+        {
+            var from = DateTime.Parse(fromDate).Date;
+            var to = DateTime.Parse(toDate).Date.AddDays(1).AddTicks(-1);
+            var fromLedgerPeriod = this.ledgerPeriodPack.GetPeriodNumber(from);
+            var toLedgerPeriod = this.ledgerPeriodPack.GetPeriodNumber(to);
+
+            var supplierSpends = this.spendsRepository.FilterBy(
+                    x =>
+                        x.Supplier.VendorManager != null &&
+                        x.LedgerPeriod >= fromLedgerPeriod && x.LedgerPeriod <= toLedgerPeriod
+                        && (string.IsNullOrWhiteSpace(vendorManagerId)
+                            || x.Supplier.VendorManager.Id == vendorManagerId))
+                .ToList();
+
+            var vendorManagerName = "ALL";
+            if (!string.IsNullOrWhiteSpace(vendorManagerId))
+            {
+                var vendorManager = this.vendorManagerRepository.FindById(vendorManagerId);
+                vendorManagerName = $"{vendorManagerId} - {vendorManager.Employee.FullName} ({vendorManager.UserNumber})";
+            }
+
+            var reportLayout = new SimpleGridLayout(
+                this.reportingHelper,
+                CalculationValueModelType.Value,
+                null,
+                $"Spend by supplier report for Vendor Manager: {vendorManagerName} between ledger period {fromLedgerPeriod} to {toLedgerPeriod}.");
+
+            AddSupplierByDateRangeReportColumns(reportLayout);
+
+            var values = new List<CalculationValueModel>();
+
+            var distinctSupplierSpends = supplierSpends.DistinctBy(x => x.SupplierId).Select(
+                x => new SupplierSpendWithTotals
+                         {
+                             BaseTotal = x.BaseTotal.HasValue ? x.BaseTotal.Value : 0,
```

Comment:
.GetValueOrDefault() (Or megrging the conditional) is cleaner for all of these nullable fields 'value or zero' style checks